zerver/lib/cache.py

```diff
@@ -69,18 +76,21 @@ def get_or_create_key_prefix():
 
     return prefix
 
-KEY_PREFIX = get_or_create_key_prefix()
+KEY_PREFIX = get_or_create_key_prefix() # type: str
 
 def bounce_key_prefix_for_testing(test_name):
+    # type: (str) -> None
     global KEY_PREFIX
     KEY_PREFIX = test_name + ':' + str(os.getpid()) + ':'
 
 def get_cache_backend(cache_name):
+    # type: (str) -> get_cache
     if cache_name is None:
         return djcache
     return get_cache(cache_name)
 
 def cache_with_key(keyfunc, cache_name=None, timeout=None, with_statsd_key=None):
+    # type: ignore # TODO: https://www.python.org/dev/peps/pep-0484/#callable
```

Comment:
Why is this a `type: ignore`?  In general I've been going with a strategy of only using `type: ignore` where there's a mypy bug.  We should either not annotate this or annotate it correctly.  

If you're having trouble figuring out how to annotate this properly, let me know and I can help!